test/irmin-pack/test_pack.ml

```diff
@@ -148,10 +148,11 @@ module Dict = struct
       Alcotest.(check (option string)) k None (Dict.find d2.dict i)
     in
     let check_raise k =
+      let open Irmin_pack_unix.Io in
       try
         ignore_int (Dict.index d2.dict k);
         Alcotest.fail "RO dict should not be writable"
-      with Irmin_pack.RO_not_allowed -> ()
+      with Irmin_pack.RO_not_allowed | Io_error `Ro_not_allowed -> ()
```

Comment:
this will be handled differently by https://github.com/mirage/irmin/pull/1878 - I propose we merge #1878 first